byteparse/shared/src/main/scala/byteparse/classparse/ClassParser.scala

```diff
@@ -0,0 +1,456 @@
+        classInfo.interfacesIndexes.map(idx =>
+          Class(classInfo, classInfo.getInfoByIndex[ClassInfo](idx).get)),
+        classInfo.fields.map(getFieldFromInfo),
+        classInfo.methods.map(getMethodFromInfo),
+        classInfo.attributes.map(attr => convertToAttribute(classInfo, attr))
+      )
+
+    }
+  }
+
+  import Info._
+  import BasicElems._
+  import ByteUtils.BE._
+
+  val constantClassInfo =
+    P( BS(7) ~ AnyWordI /*name_index*/)
+     .map(ClassInfo)
+
+  val constantFieldRefInfo =
+    P( BS(9) ~
+       AnyWordI /*class_index*/ ~
+       AnyWordI /*name_and_type_index*/
+     ).map(FieldRefInfo.tupled)
+
+  val constantMethodRefInfo =
+    P( BS(10) ~
+      AnyWordI /*class_index*/ ~
+      AnyWordI /*name_and_type_index*/
+    ).map(MethodRefInfo.tupled)
+
+  val constantInterfaceMethodRefInfo =
+    P( BS(11) ~
+       AnyWordI /*class_index*/ ~
+       AnyWordI /*name_and_type_index*/
+     ).map(InterfaceMethodRefInfo.tupled)
+
+  val constantStringInfo =
+    P( BS(8) ~ AnyWordI /*string_index*/)
+      .map(StringInfo)
+
+  val constantIntInfo = P( BS(3) ~ AnyDword.! ).map(bs =>
+    BasicElemInfo(IntElem(wrapByteBuffer(bs).getInt)))
+  val constantFloatInfo = P( BS(4) ~ AnyDword.! ).map(bs =>
+    BasicElemInfo(FloatElem(wrapByteBuffer(bs).getFloat)))
+  val constantLongInfo = P( BS(5) ~ AnyByte.rep(exactly=8).! ).map(bs =>
+    BasicElemInfo(LongElem(wrapByteBuffer(bs).getLong)))
+  val constantDoubleInfo = P( BS(6) ~ AnyByte.rep(exactly=8).! ).map(bs =>
+    BasicElemInfo(DoubleElem(wrapByteBuffer(bs).getDouble)))
+
+  val constantNameAndTypeInfo =
+    P( BS(12) ~
+       AnyWordI /*name_index*/ ~
+       AnyWordI /*descriptor_index*/
+    ).map(NameAndTypeInfo.tupled)
+
+  val constantUtf8Info =
+    P( BS(1) ~
+       AnyWordI.flatMap(l =>
+         P( AnyByte.rep(exactly=l).! ))
+    ).map(Utf8Info)
+
+  val constantMethodHandleInfo =
+    P( BS(15) ~
+       AnyByte.! /*reference_kind*/ ~
+       AnyWordI /*reference_index*/
+    ).map {
+      case (refKind, refIdx) => MethodHandleInfo(refKind(0).toInt, refIdx)
+    }
+
+  val constantMethodTypeInfo =
+    P( BS(16) ~ AnyWordI /*descriptor_index*/ )
+      .map(MethodTypeInfo)
+
+  val constantInvokeDynamicInfo =
+    P( BS(18) ~
+       AnyWordI /*bootstrap_method_attr_index*/ ~
+       AnyWordI /*name_and_type_index*/
+     ).map(InvokeDynamicInfo.tupled)
+
+  val singleConstantPoolItem = P( constantClassInfo      | constantFieldRefInfo |
+                                  constantMethodRefInfo  | constantInterfaceMethodRefInfo |
+                                  constantStringInfo     | constantIntInfo |
+                                  constantFloatInfo      | constantNameAndTypeInfo |
+                                  constantUtf8Info       | constantMethodHandleInfo |
+                                  constantMethodTypeInfo | constantInvokeDynamicInfo )
```

Comment:
In general, this is the only sort of parser which I'd say is OK to be spread over several lines: where it's already basically as compact as possible and it's only multi-line because of how many identical cases there are. 

For most other parsers, if it's multi-lined with a bunch of heterogenous different things inside, we should try to extract those into local `val`s and keep the individual parsers single-line if possible